compiler/src/dotty/tools/dotc/typer/Synthesizer.scala

```diff
@@ -279,7 +279,7 @@ class Synthesizer(typer: Typer)(using @constructorOnly c: Context):
             case mirroredType: HKTypeLambda =>
               val elems =
                 mirroredType.derivedLambdaType(
-                  resType = TypeOps.nestedPairs(accessors.map(mirroredType.memberInfo(_).widenExpr))
```

Comment:
`mirroredType.memberInfo(_)` makes no sense here. `mirroredType` is a type lambda, and as such not a legal prefix on which we can take a memberInfo. So I changed to just `_.info` which avoids the crash. However, I have no idea whether  that's correct.